router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -0,0 +1,179 @@
+	loggerNm    string
+)
+
+const (
+	DISABLE_DEST  = "DISABLE_DESTINATION"
+	REFRESH_TOKEN = "REFRESH_TOKEN"
+)
+
+// The response from the transformer network layer will be sent with a output property(in case of an error)
+type ErrorOutput struct {
+	Output ErrorResponse `json:"output"`
+}
+
+// This struct represents the datastructure present in Transformer network layer Error builder
+type ErrorResponse struct {
+	Message           string                 `json:"message"`
+	Destination       map[string]interface{} `json:"destination"`
+	NetworkFailure    bool                   `json:"networkFailure"`
+	Status            int                    `json:"status"`
+	AuthErrorCategory string                 `json:"authErrorCategory"`
+	AccessToken       string                 `json:"accessToken"`
+}
+
+type RefreshTokenBody struct {
+	hasExpired   bool
+	expiredToken string
+}
+
+// Custom Marshalling for RefreshTokenBody struct
+func (refTokenBody RefreshTokenBody) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(struct {
+		HasExpired   bool
+		ExpiredToken string
+	}{
+		HasExpired:   refTokenBody.hasExpired,
+		ExpiredToken: refTokenBody.expiredToken,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return j, nil
+}
+
+func init() {
+	configBEURL = backendconfig.GetConfigBackendURL()
+	pkgLogger = logger.NewLogger().Child("router").Child("OAuthResponseHandler")
+	loggerNm = "OAuthResponseHandler"
+}
+
+func (authErrHandler *OAuthErrResHandler) Setup() {
+	authErrHandler.logger = pkgLogger
+	authErrHandler.tr = &http.Transport{}
+	//This timeout is kind of modifiable & it seemed like 10 mins for this is too much!
+	authErrHandler.client = &http.Client{Transport: authErrHandler.tr, Timeout: 5 * time.Minute}
+	authErrHandler.oauthErrHandlerReqTimerStat = stats.NewStat("router.processor.oauthErrorHandler_request_time", stats.TimerType)
+	authErrHandler.oauthErrHandlerNetReqTimerStat = stats.NewStat("router.oauthErrorHandler_network_request_time", stats.TimerType)
+}
+
+func (authErrHandler *OAuthErrResHandler) RefreshToken(workspaceId string, accountId string, accessToken string) (statusCode int, respBody string) {
+	authErrHandler.oauthErrHandlerReqTimerStat.Start()
+	if len(strings.TrimSpace(accessToken)) == 0 {
+		return http.StatusBadRequest, `Cannot proceed with refresh token request as accessToken is empty`
+	}
+	authErrHandler.oauthErrHandlerNetReqTimerStat.Start()
+	refreshUrl := fmt.Sprintf("%s/dest/workspaces/%s/accounts/%s/token", configBEURL, workspaceId, accountId)
+	authErrHandler.oauthErrHandlerNetReqTimerStat.End()
+	refTokenBody := RefreshTokenBody{
+		hasExpired:   true,
+		expiredToken: accessToken,
+	}
+	res, err := json.Marshal(refTokenBody)
+	if err != nil {
+		// TODO: Is this way ok ? - Check with team
+		return http.StatusBadRequest, err.Error()
+	}
+	resp, refErr := authErrHandler.client.Post(refreshUrl, "application/json; charset=utf-8", bytes.NewBuffer(res))
+	if refErr != nil {
+		panic(refErr)
```

Comment:
Maybe we should throw some 5XX error from here as we are refreshing at this poin